Source/Plugins/Core/com.equella.core/js/tsrc/search/components/RefineSearchPanel.tsx

```diff
@@ -52,31 +55,81 @@ interface RefinePanelProps {
    * Child components rendered inside this panel.
    */
   controls: RefinePanelControl[];
+  /**
+   * Handler for when the show more/show less button is clicked
+   */
+  handleClick: (expansionState: boolean) => void;
+  /**
+   * Determines the current state of the filter expansion panel
+   */
+  panelExpanded: boolean;
+  /**
+   * True if filter icon should be shown next to the 'Show more' button
+   */
+  showFilterIcon: boolean;
 }
 
-export const RefineSearchPanel = ({ controls }: RefinePanelProps) => {
+export const RefineSearchPanel = ({
+  controls,
+  handleClick,
+  panelExpanded,
+  showFilterIcon,
+}: RefinePanelProps) => {
   const { title } = languageStrings.searchpage.refineSearchPanel;
+
+  const { showmore, showless } = languageStrings.common.action;
+
+  const [collectionControl, ...collapsedControls] = controls.filter(
```

Comment:
There's an assumption being made here that the Collection control is first.

To some degree, that's outside the purview of this component. The contract for this component is that it will display the **first** component outside of the selection control, so perhaps this should just be:

```suggestion
  const [alwaysVisibleControl, ...collapsedControls] = controls.filter(
```

For now that's all we need. But if that ever changed then we'd possibly allow two lots of controls to be passed in - those in the expansion panel, those outside of it.

Reply:
Yeah, I had originally envisioned to pass in two sets of controls, but yeah, we can cross that bridge if the need arises in the future